src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -135,25 +136,25 @@ internal Guid GetTelemetryId(FixAllScope? fixAllScope = null)
         /// The sequence of operations that define the code action.
         /// </summary>
         public Task<ImmutableArray<CodeActionOperation>> GetOperationsAsync(CancellationToken cancellationToken)
-            => GetOperationsAsync(new ProgressTracker(), cancellationToken);
+            => GetOperationsAsync(originalSolution: null!, new ProgressTracker(), cancellationToken);
```

Comment:
we can consider that.  generally speaking though that's not been what we've done with anything else.  We'd also potentially want to really rethink these apis overall (they're terrible).

Reply:
>  want to really rethink these apis overall (they're terrible).

+1 on that. I don't think we should be introducing a new public API that address only this issue. The CodeAction APIs have multiple other issues as well that need new API.

Reply:
> Otherwise any custom implementations still will have bugs, right?

Depends on the workspace _and_ the type of code action. Code actions that just change code won't have problems. Code actions in workspaces that aren't updated in parallel with the code action application also aren't affected.